src/graph.h

```diff
@@ -91,10 +91,14 @@ class ElementRef {
 // and can be passed as bound inputs to Simit programs.
 class Set {
 public:
-  Set(const std::string &name)
-      : name(name), numElements(0), endpoints(nullptr),
+  enum Kind {Unstructured, LatticeLink};
+  
+  Set(const std::string &name, Kind kind=Unstructured)
```

Comment:
I moved this constructor to Private, and reintroduced the name-only constructor with delegation to this one. Keeping this around makes it easier to delegate from the Lattice Link constructors.